dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataLogsConnector.java

```diff
@@ -118,7 +118,11 @@ public void addTasksTo(List<? super Task<?>> out, @Nonnull ConnectorArguments ar
     // Beware of Teradata SQLSTATE HY000. See issue #4126.
     // Most likely caused by some operation (equality?) being performed on a datum which is too long
     // for a varchar.
-    ZonedIntervalIterable intervals = ZonedIntervalIterable.forConnectorArguments(arguments);
+    ZonedIntervalIterable intervals =
+        isAssessment
+            ? ZonedIntervalIterable.forConnectorArguments(
+                arguments, arguments.getQueryLogRotationFrequency())
+            : ZonedIntervalIterable.forConnectorArguments(arguments);
```

Comment:
Can we please not have a bifurcation of functionality here. I'd like reviewers to press more for single functionality where possible.